### ISSUE TITLE
Reproduced Accuracy

### ISSUE THREAD

**Issue by MrChenFeng** [AndrewAtanov/simclr-pytorch#2]
Hi, Andrews,

Thanks so much for sharing your implementation.
Your result is the most close one I've ever seen about SimCLR. I do have a question, how important it is to use lbfgs logistic regression rather than a normal classifer in evaluation? did it make a big change?

**Comment by AndrewAtanov** [comment]
Hi Chen,

Thanks for your interest. In our implementation, we use simple SGD to train a linear classifier (see [this line](https://github.com/AndrewAtanov/simclr-pytorch/blob/71fd5469af93bf4e5d15819bd8056f5bc23b4c15/models/ssl.py#L344) for the reference) and didn't try to use L-BFGS.

**Comment by MrChenFeng** [comment]
Hi Andrew,
Thanks so much for your quick reply.
In my own implementation, I can only get 91.5% with 512 batch and 1000 epochs. What do think is the necessary module to reproduce the result? Or you get the 93.5% in your first try?

**Comment by senya-ashukha** [comment]
Hi folks,

Haha, not in the first run definitely, but in the end is stable :)

@AndrewAtanov correct me if I'm wrong but it seems that we used BS 1024 for both self-supervised [1] training and linear [2] eval. Because of two augmentations per image real BS for self-supervised pre-training ends up 2048 (1024 images, 2 augmentations each).

As far as I remember, LARS and worm-up of learning rate were really important.

[1] https://github.com/AndrewAtanov/simclr-pytorch/blob/master/configs/cifar_train_epochs1000_bs1024.yaml
[2] https://github.com/AndrewAtanov/simclr-pytorch/blob/master/configs/cifar_eval.yaml

**Comment by MrChenFeng** [comment]
Hi Senya, 
Thanks so much for your detailed suggestions! I will try it now.
